### ISSUE TITLE
Make button text change when state chagnes

### ISSUE THREAD

**Issue by xescugc** [mcarpenter622/ebitenui#160]
So I want to do something like 
![button_normal](https://github.com/ebitenui/ebitenui/assets/1532900/cf111d50-0f00-4e8e-b46d-533e24e42fed)

I opened a [PR](https://github.com/ebitenui/ebitenui/pull/159) for the color of the text, that is fine. But I was wondering which would be the approach to make the text of the button also move down when pressed.

I was thinking on adding a `IsPressed` (similar to the [`IsFocused`](https://pkg.go.dev/github.com/ebitenui/ebitenui/widget#Button.IsFocused)) and then, somehow, change the text position. My only guess would be to expose the [`ButtonOption.TextPosittion`](https://pkg.go.dev/github.com/ebitenui/ebitenui/widget#ButtonOptions.TextPosition) so it can be changed thorough the button directly but that would not even work also as it's a `TextPosition` which is `Start|Center|End`.

The other option is to go with the [`ButtonOption.TextPading`](https://pkg.go.dev/github.com/ebitenui/ebitenui/widget#ButtonOptions.TextPadding) so I would need to expose that be able to change it.

The ultimate end solution is to write the text on the image of the button myself with an [`ebiten/text.Draw()`](https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2@v2.7.6/text#Draw) but I would rather add the possibility to EbitenUI if possible hehe.

Which would be the right way to do it?

**Comment by xescugc** [comment]
So to try to do this I added a handler on the click to try to change the `button.Text().Inset` but for what I've seen the only change it actually applies is the `Insets.Left` and `Insets.Right` the `Insets.Top` and `Insets.Bottom` are ignored.

I checked a bit and for what I see the `Top` and `Bottom` are only used for the container on the Button to render the text (with the `Right` and `Left`) but inside the `NewText` only the `Left` and `Right` are used.

https://github.com/xescugc/ebitenui/commit/c76612a65c5f01a18a741b2604e56af04c7f29fb

So I still don't know how to do/fix this, any ideas @mcarpenter622 ?

TL;DR:

Want to change the Y position of the text when clicked

**Comment by mcarpenter622** [comment]
Please check out the button demo in the latest commit. This should be do-able now.
https://github.com/ebitenui/ebitenui/commit/f5293384011553aea4b3a6b8d49da61f56fc1cc3

**Comment by xescugc** [comment]
Perfect it works amazingly!
